src/main/java/hudson/plugins/logparser/LogParserAction.java

```diff
@@ -1,232 +1,230 @@
-            }
-
-            @Override
-            public String generateToolTip(CategoryDataset dataset, int row,
-                    int column) {
-                switch (row) {
-                case 0:
-                    return "Errors: " + result.getTotalErrors();
-                case 1:
-                    return "Warnings: " + result.getTotalWarnings();
-                default:
-                    return "Infos: " + result.getTotalInfos();
-                }
-            }
-        };
-        plot.setRenderer(ar);
-        ar.setSeriesPaint(0, ColorPalette.RED);    // error
-        ar.setSeriesPaint(1, ColorPalette.BLUE);   // info
-        ar.setSeriesPaint(2, ColorPalette.YELLOW); // warning
-
-        // crop extra space around the graph
-        plot.setInsets(new RectangleInsets(0, 0, 0, 5.0));
-
-        return chart;
-    }
-
-    private String getRelPath(StaplerRequest req) {
-        String relPath = req.getParameter("rel");
-        if (relPath == null)
-            return "";
-        return relPath;
-    }
-
-}
+package hudson.plugins.logparser;
+
+import hudson.Functions;
+import hudson.model.Action;
+import hudson.model.AbstractBuild;
+import hudson.model.Run;
+import hudson.util.Area;
+import hudson.util.ChartUtil;
+import hudson.util.ColorPalette;
+import hudson.util.DataSetBuilder;
+import hudson.util.ShiftedCategoryAxis;
+import hudson.util.StackedAreaRenderer2;
+
+import java.awt.Color;
+import java.io.File;
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+
+import org.jfree.chart.ChartFactory;
+import org.jfree.chart.JFreeChart;
+import org.jfree.chart.axis.CategoryAxis;
+import org.jfree.chart.axis.CategoryLabelPositions;
+import org.jfree.chart.axis.NumberAxis;
+import org.jfree.chart.plot.CategoryPlot;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.chart.renderer.category.StackedAreaRenderer;
+import org.jfree.data.category.CategoryDataset;
+import org.jfree.ui.RectangleInsets;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+
+public class LogParserAction implements Action {
+
+    final private Run<?, ?> build;
+    final private LogParserResult result;
+
+    private static String urlName = "parsed_console";
+
+    @Deprecated
+    public LogParserAction(final AbstractBuild<?, ?> build, final LogParserResult result) {
+        this((Run<?, ?>) build, result);
+    }
+
+    public LogParserAction(final Run<?, ?> build, final LogParserResult result) {
+        this.build = build;
+        this.result = result;
+    }
+
+    public String getIconFileName() {
+        return "clipboard.gif";
+    }
+
+    public String getDisplayName() {
+        return "Parsed Console Output";
+    }
+
+    public String getUrlName() {
+        return urlName;
+    }
+
+    public static String getUrlNameStat() {
+        return urlName;
+    }
+
+    public Run<?, ?> getOwner() {
```

Comment:
`@WithBridgeMethods` should be used here (something like [this](https://github.com/jenkinsci/analysis-core-plugin/blob/42c50ab8f5cca9da0d83519a03afd7fe2a46de57/src/main/java/hudson/plugins/analysis/views/ConsoleDetail.java#L114)).